Jenkinsfile

```diff
@@ -169,43 +187,33 @@ pipeline {
                             deleteDir()
                             unstash 'CIScripts'
                             powershell script: './CIScripts/Test.ps1'
+                            stash name: 'testReport', includes: 'testReport.xml'
                         }
                     }
                 }
             }
         }
     }
 
-    environment {
-        LOG_SERVER = "logs.opencontrail.org"
-        LOG_SERVER_USER = "zuul-win"
-        LOG_ROOT_DIR = "/var/www/logs/winci"
-        BUILD_SPECIFIC_DIR = "${ZUUL_UUID}"
-        JOB_SUBPATH = env.JOB_NAME.replaceAll("/", "/job/")
-        REMOTE_DST_FILE = "${LOG_ROOT_DIR}/${BUILD_SPECIFIC_DIR}/log.txt.gz"
-    }
-
     post {
         always {
             node('master') {
                 script {
-                    // Job triggered by Zuul -> upload log file to public server.
-                    // Job triggered by Github CI repository (variable "ghprbPullId" exists) -> keep log "private".
-
-                    // TODO JUNIPER_WINDOWSSTUBS variable check is temporary and should be removed once
-                    // repository contrail-windows is accessible from Gerrit and it is main source of
-                    // windowsstubs code.
-                    if (env.ghprbPullId == null && env.JUNIPER_WINDOWSSTUBS == null) {
-                        // unstash "buildLogs"
-                        // TODO correct flags for rsync
-                        sh "ssh ${LOG_SERVER_USER}@${LOG_SERVER} \"mkdir -p ${LOG_ROOT_DIR}/${BUILD_SPECIFIC_DIR}\""
-                        def tmpLogFilename = "clean_log.txt.gz"
-                        obtainLogFile(env.JOB_NAME, env.BUILD_ID, tmpLogFilename)
-                        sh "rsync $tmpLogFilename ${LOG_SERVER_USER}@${LOG_SERVER}:${REMOTE_DST_FILE}"
-                        deleteDir()
+                    try {
+                        unstash 'testReport'
+                        publishToLogServer('testReport.xml', logsDestination)
+                    } catch (err) {
+                        // unstash may throw an exception when pipeline failed before Test stage,
+                        // but we still need logs/metrics to be pushed
                     }
                 }
 
+                script {
+                    def logFilename = "log.txt.gz"
+                    obtainLogFile(env.JOB_NAME, env.BUILD_ID, logFilename)
+                    publishToLogServer(logFilename, logsDestination)
```

Comment:
There's some meh duplication here with multiple calls to publishToLogServer. not sure how to best get rid of it though